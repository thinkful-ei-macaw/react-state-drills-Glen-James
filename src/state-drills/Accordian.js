import React, { Component } from "react";
import renderer from 'react-test-renderer';

class Accordian extends Component {

    constructor(props) {
        super(props)

        this.state = {

            currentSelected: -1
        }
    }

    buttonClick = (index) => {

        this.setState({

            currentSelected: index


        })


    }

    
   
    render() {
        let sections;
        if(!this.props.sections){
             sections = <li></li>          
          
        }else{

          sections = this.props.sections.map((section, index) => {

                if(index === this.state.currentSelected){
                    return (
                        <li key={index}>
                            <button onClick={() => this.buttonClick(-1)}>Click Me</button>
                            <p>{section.title}</p>
                            <p>{section.content}</p>
                        </li>     
                    )
                }
                    return (
                        <li key={index}>
                            <button onClick={() => this.buttonClick(index)}>{section.title}</button>
                        </li>  
                    )
                        
            })


        }

        


        return (
            <div>
            <ul>
               {sections}
            </ul>
            </div>
        );
    }


}


export default Accordian;