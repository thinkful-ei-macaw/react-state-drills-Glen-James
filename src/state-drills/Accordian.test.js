import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';


//SMOKSCREEN TEST

it('renders with no crash', () => {

    const div = document.createElement('div');
    ReactDOM.render(<Accordian sections={[{title: 'title', content: 'content' }]}/>, div);
    ReactDOM.unmountComponentAtNode(div);


})

//SNAPSHOT TEST

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Accordian sections={[{title: 'title', content: 'content' }]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//SNAPSHOT TEST EMPTY LI

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Accordian />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

