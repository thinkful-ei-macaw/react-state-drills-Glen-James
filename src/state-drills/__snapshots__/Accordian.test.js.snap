// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        title
      </button>
    </li>
  </ul>
</div>
`;

exports[`renders the UI as expected 2`] = `
<div>
  <ul>
    <li />
  </ul>
</div>
`;
